api/tests/unit/infrastructure/repositories/challenge-repository_test.js

```diff
@@ -338,88 +338,103 @@ describe('Unit | Repository | challenge-repository', () => {
       });
 
       context('when query happens with no error', () => {
-
-        let promise;
-
-        beforeEach(() => {
+        it('returns challenges', async () => {
```

Comment:
J'ai du mal à percevoir la plus-value de ce refactoring. L'information me parait moins claire. Est-ce que tu peux m'expliquer l'intention derrière cette action stp ?

Reply:
Si on reprend les expects des tests tels qu'ils était pre-refacto :

`expect(promise).to.be.fulfilled`
C'est redondant avec les tests suivants, si t'arrives à faire un `expect` sur le résultat c'est que la promesse est résolue. C'est un test de construction (échafaudage) intéressant pendant l'implémentation dans une démarche itérative TDD, mais ce n'est pas un test qu'il est intéressant de garder une fois l'implémentation finie à mon sens.

`expect(challengeDatasource.findOperative).to.have.been.calledWithExactly()`
Ici `challengeDatasource.findOperative()` est une `Query` au sens où cette méthode retourne quelque chose par opposition à une `Command` (qui elle ne retournerait rien comme par exemple `log(message)`). Lorsqu'on cherche à tester qu'une `Command` a bien éte envoyée, on n'a d'autres choix que de vérifier qu'on l'a bien appelée avec les bons arguments. Dans le cas d'une `Query`, on va plutôt tester que tout se passe bien dans l'utilisation de cette dépendance via l'effet qu'elle a sur le `resultat` final de la méthode testée. En effet si on récupère de l'information via cette `Query`, c'est pour s'en servir et donc que le résultat globale en dépend.
Voir ici : https://www.youtube.com/watch?v=URSWYvyc42M
Et là : https://1024pix.atlassian.net/wiki/spaces/DEV/pages/600178695/Tester

```
expect(challenges).to.be.an('array');	
expect(challenges).to.have.lengthOf(2);	
challenges.map((challenge) => expect(challenge).to.be.an.instanceOf(Challenge))
```
Ici, on est beaucoup trop light en terme de vérification, certes on va avoir un tableau contenant 2 challenges, mais qui nous dit qu'ils contiendront les bonnes informations ? (surtout en JS, où je peux instancier n'importe quoi n'importe comment sans que rien ne bronche)

`expect(challenges[0].skills).to.deep.equal([...])`
Ce test est l'un des plus intéressants, j'ai repris cette philosophie pour tester l'entièreté du retour de la méthode et pas seulement les skills.